Bug when running SwiftLint on a non-root directory

### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Running SwiftLint directly on a module within my codebase (directory in the tree) only uses the configuration file in that directory instead of merging it with the configuration file at the root.

I would expect this to have the same behavior as running SwiftLint at the root does. In that case, the root configuration file is merged with the configuration file in my module, and everything works as expected.

**Sample project:** [DirectoryBug.zip](https://github.com/realm/SwiftLint/files/5398412/DirectoryBug.zip)


##### Complete output when running SwiftLint, including the stack trace and command used

`swiftlint lint --no-cache`

```
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths 
Linting 'ViewController.swift' (1/4)
Linting 'AppDelegate.swift' (2/4)
Linting 'SceneDelegate.swift' (3/4)
Linting 'MyType.swift' (4/4)
Done linting! Found 0 violations, 0 serious in 4 files.
Program ended with exit code: 0
```

With working directory set to the root of the DirectoryBug project:
`swiftlint lint --no-cache -- DirectoryBug/Module1`

```
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths DirectoryBug/Module1
Linting 'MyType.swift' (1/1)
/Users/sfriedman/Desktop/DirectoryBug/DirectoryBug/Module1/MyType.swift:10:51: warning: Unneeded NotificationCenter Removal Violation: Observers are automatically unregistered on dealloc (iOS 9 / macOS 10.11) so you should't call `removeObserver(self)` in the deinit. (unneeded_notification_center_removal)
Done linting! Found 1 violation, 0 serious in 1 file.
Program ended with exit code: 0
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? `master`
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Building from source
* Paste your configuration file:

See attached sample project

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
  Yes, `.swiftlint.yml` at the root of the project and `DirectoryBug/Module1/.swiftlint.yml`
* Which Xcode version are you using (check `xcodebuild -version`)? 11.7
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

Yes, see description

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -112,6 +112,11 @@
 * Fix finding a nested config when a single file path is passed.  
   [Seth Friedman](https://github.com/sethfri)
 
+* Fix incorrect calculation of the root path when a directory in the
+  tree is passed in as a path argument.  
+  [Seth Friedman](https://github.com/sethfri)
+  [#3383](https://github.com/realm/SwiftLint/issues/3383)
+
 ## 0.40.3: Greased Up Drum Bearings
 
 #### Breaking

--- a/Source/swiftlint/Extensions/Configuration+CommandLine.swift
+++ b/Source/swiftlint/Extensions/Configuration+CommandLine.swift
@@ -227,16 +227,12 @@ extension Configuration {
         })
     }
 
-    private static func rootPath(from paths: [String]) -> String? {
-        // We don't know the root when more than one path is passed (i.e. not useful if the root of 2 paths is ~)
-        return paths.count == 1 ? paths.first?.absolutePathStandardized() : nil
-    }
-
     // MARK: LintOrAnalyze Command
 
     init(options: LintOrAnalyzeOptions) {
         let cachePath = options.cachePath.isEmpty ? nil : options.cachePath
-        self.init(path: options.configurationFile, rootPath: Self.rootPath(from: options.paths),
+        self.init(path: options.configurationFile,
+                  rootPath: FileManager.default.currentDirectoryPath.bridge().absolutePathStandardized(),
                   optional: isConfigOptional(), quiet: options.quiet, enableAllRules: options.enableAllRules,
                   cachePath: cachePath)
     }
@@ -256,7 +252,8 @@ extension Configuration {
 
     init(options: AutoCorrectOptions) {
         let cachePath = options.cachePath.isEmpty ? nil : options.cachePath
-        self.init(path: options.configurationFile, rootPath: Self.rootPath(from: options.paths),
+        self.init(path: options.configurationFile,
+                  rootPath: FileManager.default.currentDirectoryPath.bridge().absolutePathStandardized(),
                   optional: isConfigOptional(), quiet: options.quiet, cachePath: cachePath)
     }
 

